puffin/src/lib.rs

```diff
@@ -718,9 +814,23 @@ macro_rules! profile_function {
     };
     ($data:expr) => {
         let _profiler_scope = if $crate::are_scopes_on() {
+            static mut _FUNCTION_NAME: &'static str = "";
+            static mut _LOCATION: &'static str = "";
+            static _INITITIALIZED: ::std::sync::Once = ::std::sync::Once::new();
```

Comment:
would it make sense to use thread local here to avoid the synchronization?

Reply:
I tried using `thread_local`, and performance seems to be maybe 1-2% worse, but maybe within noise, and code is much nicer. However, it means duplicating the string for each thread, and also it won't work once we implement https://github.com/EmbarkStudios/puffin/issues/167, so I think this slightly more verbose version is better for now